pkg/pool-weighted/test/foundry/LiquidityApproximationWeighted.t.sol

```diff
@@ -88,413 +90,184 @@ contract LiquidityApproximationWeightedTest is LiquidityApproximationTest, Weigh
-        swapFeePercentage = bound(swapFeePercentage, minSwapFeePercentage, maxSwapFeePercentage);
-        weightDai = bound(weightDai, 1e16, 99e16);
-        daiAmountIn = bound(daiAmountIn, minAmount, _computeMaxTokenAmount(weightDai));
-
-        addLiquidityUnbalancedWeights(daiAmountIn, swapFeePercentage, weightDai);
-    }
-
-    function testAddLiquidityUnbalancedWeightsNoSwapFee__Fuzz(uint256 daiAmountIn, uint256 weightDai) public {
+    function fuzzPoolParams(uint256[POOL_SPECIFIC_PARAMS_SIZE] memory params) internal override {
+        uint256 weightDai = params[0];
         weightDai = bound(weightDai, 1e16, 99e16);
-        daiAmountIn = _computeMaxTokenAmount(weightDai);
 
-        addLiquidityUnbalancedWeights(daiAmountIn, 0, weightDai);
-    }
-
-    function testAddLiquidityUnbalancedWeightsSmallAmounts__Fuzz(uint256 daiAmountIn, uint256 weightDai) public {
-        daiAmountIn = bound(daiAmountIn, 1, 1e6);
-        weightDai = bound(weightDai, 1e16, 99e16);
+        _setPoolBalancesWithDifferentWeights(weightDai);
 
-        // For small amounts, BPT amount out goes negative because of rounding and BasePoolMath reverts.
-        // Perform an external call so that `expectRevert` catches the error.
-        vm.expectRevert(stdError.arithmeticError);
-        this.addLiquidityUnbalancedWeights(daiAmountIn, 0, weightDai);
-    }
+        absoluteRoundingDelta = 1e14;
 
-    function addLiquidityUnbalancedWeights(uint256 daiAmountIn, uint256 swapFeePercentage, uint256 weightDai) public {
         // Weights can introduce some differences in the swap fees calculated by the pool during unbalanced add/remove
         // liquidity, so the error tolerance needs to be a bit higher than the default tolerance.
-        excessRoundingDelta = 0.5e16; // 0.5%
-
-        _setPoolBalancesWithDifferentWeights(weightDai);
-
-        uint256 amountOut = addUnbalancedOnlyDai(daiAmountIn, swapFeePercentage);
-        swapFeePercentage > 0
-            ? assertLiquidityOperation(amountOut, swapFeePercentage, true)
-            : assertLiquidityOperationNoSwapFee();
-    }
-
-    function testAddLiquiditySingleTokenExactOutWeights__Fuzz(
-        uint256 exactBptAmountOut,
-        uint256 swapFeePercentage,
-        uint256 weightDai
-    ) public {
-        // exactBptAmountOut = bound(exactBptAmountOut, minAmount, maxAmount / 2 - 1);
-        swapFeePercentage = bound(swapFeePercentage, minSwapFeePercentage, maxSwapFeePercentage);
-        weightDai = bound(weightDai, 1e16, 99e16);
-        exactBptAmountOut = bound(exactBptAmountOut, minAmount, _computeMaxBptAmount(weightDai, swapFeePercentage));
+        excessRoundingDelta = 2e16; // 2%
+        defectRoundingDelta = 0.001e16; // 0.001%
 
-        _addLiquiditySingleTokenExactOutWeights(exactBptAmountOut, swapFeePercentage, weightDai);
+        if (weightDai < 20e16 || weightDai > 80e16) {
+            // TODO: check why the difference grows so large in the extremes (high fees, high dai weight).
```

Comment:
Was this intended to be resolved in this PR? If not, can just leave it.

Reply:
Not really. The idea of this PR was to allow fuzzing specific pool params without the need to rewrite the Liquidity Approximation tests. So, it's out of scope to evaluate the math of the weighted pool. This "To Do" comment was in the code before, but in another place.